wp-job-manager-functions.php

```diff
@@ -212,15 +212,29 @@ function get_job_listings_keyword_search( $args ) {
 		// Search taxonomy
 		$conditions[] = "{$wpdb->posts}.ID IN ( SELECT object_id FROM {$wpdb->term_relationships} AS tr LEFT JOIN {$wpdb->term_taxonomy} AS tt ON tr.term_taxonomy_id = tt.term_taxonomy_id LEFT JOIN {$wpdb->terms} AS t ON tt.term_id = t.term_id WHERE t.name LIKE '%" . esc_sql( $job_manager_keyword ) . "%' )";
 
-		if ( ctype_alnum( $job_manager_keyword ) ) {
-			$conditions[] = "{$wpdb->posts}.post_content RLIKE '[[:<:]]" . esc_sql( $job_manager_keyword ) . "[[:>:]]'";
-		} else {
-			$conditions[] = "{$wpdb->posts}.post_content LIKE '%" . esc_sql( $job_manager_keyword ) . "%'";
+		/**
+		 * Filters the conditions to use when querying job listings. Resulting array is joined with OR statements.
+		 *
+		 * @since 1.26.0
+		 *
+		 * @param array  $conditions          Conditions to join by OR when querying job listings.
+		 * @param string $job_manager_keyword Search query.
+		 */
+		$conditions = apply_filters( 'job_listing_search_conditions', $conditions, $job_manager_keyword );
+		if ( empty( $conditions ) ) {
+			return $search;
 		}
 
-		$args['where'] .= " AND ( " . implode( ' OR ', $conditions ) . " ) ";
+		$conditions_str = implode( ' OR ', $conditions );
 
-		return $args;
+		if ( ! empty( $search ) ) {
+			$search = preg_replace( '/^ AND /', '', $search );
```

Comment:
WP Query starts its `WHERE` query segment with `1=1`. The `posts_search` filter prefaces its passed default value with ` AND `. I'm guessing they do this for more predictable chaining, but it still makes using the filter a bit clunky. Other hooked filters may have added other things after the default search query from WP core, but the assumption is that whatever is passed will start with ` AND ` or will be an empty string.